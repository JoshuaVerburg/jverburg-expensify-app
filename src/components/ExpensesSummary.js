import React from 'react';
import {connect} from 'react-redux';
import numeral from 'numeral';
import totalExpenses from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';


export const ExpensesSummary = (props) => (

  <div>
    <p>Viewing {props.expenses.length} expenses totalling {numeral(totalExpenses(props.expenses)/100).format("$0,0.00")}</p>
  </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters) 
    };
};

export default connect(mapStateToProps)(ExpensesSummary);

