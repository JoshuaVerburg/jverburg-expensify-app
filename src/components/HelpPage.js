import React from 'react';

const HelpPage = () => (
    <div>
        this is from help
    </div>
);
export default HelpPage;