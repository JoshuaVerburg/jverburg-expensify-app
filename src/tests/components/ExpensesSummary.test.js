import React from 'react';
import { shallow } from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should render ExpensesSummary with expense', () => {
    const wrapper = shallow(<ExpensesSummary expenses={ expenses }/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with one expense', () => {
    const wrapper = shallow(<ExpensesSummary expenses={ [expenses[1]] }/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with no expense', () => {
    const wrapper = shallow(<ExpensesSummary expenses={[]}/>);
    expect(wrapper).toMatchSnapshot();
});