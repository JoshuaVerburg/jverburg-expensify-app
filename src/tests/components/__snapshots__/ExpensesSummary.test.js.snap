// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with expense 1`] = `
<div>
  <p>
    Viewing 
    3
     expenses totalling 
    $1,141.95
  </p>
</div>
`;

exports[`should render ExpensesSummary with no expense 1`] = `
<div>
  <p>
    Viewing 
    0
     expenses totalling 
    $0.00
  </p>
</div>
`;

exports[`should render ExpensesSummary with one expense 1`] = `
<div>
  <p>
    Viewing 
    1
     expenses totalling 
    $1,095.00
  </p>
</div>
`;
